src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -342,6 +342,16 @@ void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion
         break;
     }
 
+    // 2.B) Clamp the movement to the mutable viewport
+    if (const auto mutableViewport = _GetMutableViewport(); targetPos > mutableViewport.BottomRightInclusive())
+    {
+        targetPos.Y = mutableViewport.BottomInclusive();
+    }
+    else if (const auto bufferSize = _activeBuffer().GetSize(); targetPos < bufferSize.Origin())
+    {
+        targetPos.Y = bufferSize.Top();

```

Comment:
This doesn't restrict you from going above the visible area when htere's a scrollbar, right?

Another quick check: should moving UP at the TOP of the screen move you to 0,0? Should moving DOWN at the BOTTOM move you to width,bottom? That is how text editors select. :)

Reply:
> should moving UP at the TOP of the screen move you to 0,0? Should moving DOWN at the BOTTOM move you to width,bottom? That is how text editors select. :)

Yeah, I wasn't sure about what the desired behavior here should be. The previous commit did that, but here I changed it to just crop the y-coordinate. I'll revert that part of the change

> A size's origin should always be zero - this is an important part of it being a size...right? Is this a complicated way to say targetPos.Y < 0?

Yes, added this in for (1) symmetry (check the top and the bottom) and (2) clamp _only_ the y-coordinate. Frankly, we don't need that if-statement at all because `UpdateSelection()`'s `MoveByX()` make sure you don't fall out of the buffer. Since I'm reverting the change in the comment above, I'll just remove this if-branch entirely.